### PR TITLE
[plugins/aws][feat] Add redshift collection support

### PR DESCRIPTION
**Pull request by meln1k**
# Description

# To-Dos

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`
- [ ] Document new or updated functionality (someengineering/resoto.com#XXXX)

# Issues Fixed

- Fixes #XXXX

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: plugins/aws/resoto_plugin_aws/collector.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
     s3,
     service_quotas,
     kinesis,
+    redshift,
```
Comment (anjafr) [inline plugins/aws/resoto_plugin_aws/collector.py:25]: looks like you need to update your base branch 💡 - also I think this list is sorted alphabetically? Same for the resources in line 48

File: plugins/aws/resoto_plugin_aws/resource/redshift.py [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+@define(eq=False, slots=False)
+class AwsRedshiftVpcEndpoint:
+    kind: ClassVar[str] = "aws_redshift_vpc_endpoint"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "vpc_endpoint_id": S("VpcEndpointId"),
+        "vpc_id": S("VpcId"),
+        "network_interfaces": S("NetworkInterfaces", default=[]) >> ForallBend(AwsRedshiftNetworkInterface.mapping),
+    }
+    vpc_endpoint_id: Optional[str] = field(default=None)
+    vpc_id: Optional[str] = field(default=None)
```
Comment (anjafr) [inline plugins/aws/resoto_plugin_aws/resource/redshift.py:37]: should this result in a `dependant_node` to the corresponding instance of `AwsEc2Vpc`?

File: plugins/aws/resoto_plugin_aws/resource/redshift.py [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+    }
+    cluster_security_group_name: Optional[str] = field(default=None)
+    status: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsRedshiftVpcSecurityGroupMembership:
+    kind: ClassVar[str] = "aws_redshift_vpc_security_group_membership"
+    mapping: ClassVar[Dict[str, Bender]] = {"vpc_security_group_id": S("VpcSecurityGroupId"), "status": S("Status")}
+    vpc_security_group_id: Optional[str] = field(default=None)
```
Comment (anjafr) [inline plugins/aws/resoto_plugin_aws/resource/redshift.py:69]: should this result in a `dependant_node` to the corresponding instance of `AwsEc2SecurityGroup`?

File: plugins/aws/resoto_plugin_aws/resource/redshift.py [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+    mapping: ClassVar[Dict[str, Bender]] = {"elastic_ip": S("ElasticIp"), "status": S("Status")}
+    elastic_ip: Optional[str] = field(default=None)
+    status: Optional[str] = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsRedshiftClusterIamRole:
+    kind: ClassVar[str] = "aws_redshift_cluster_iam_role"
+    mapping: ClassVar[Dict[str, Bender]] = {"iam_role_arn": S("IamRoleArn"), "apply_status": S("ApplyStatus")}
+    iam_role_arn: Optional[str] = field(default=None)
```
Comment (anjafr) [inline plugins/aws/resoto_plugin_aws/resource/redshift.py:220]: should this result in a `dependant_node` to the corresponding instance of `AwsIamRole`?

File: plugins/aws/resoto_plugin_aws/resource/redshift.py [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+        "redshift_endpoint": S("Endpoint") >> Bend(AwsRedshiftEndpoint.mapping),
+        "redshift_automated_snapshot_retention_period": S("AutomatedSnapshotRetentionPeriod"),
+        "redshift_manual_snapshot_retention_period": S("ManualSnapshotRetentionPeriod"),
+        "redshift_cluster_security_groups": S("ClusterSecurityGroups", default=[])
+        >> ForallBend(AwsRedshiftClusterSecurityGroupMembership.mapping),
+        "redshift_vpc_security_groups": S("VpcSecurityGroups", default=[])
+        >> ForallBend(AwsRedshiftVpcSecurityGroupMembership.mapping),
+        "redshift_cluster_parameter_groups": S("ClusterParameterGroups", default=[])
+        >> ForallBend(AwsRedshiftClusterParameterGroupStatus.mapping),
+        "redshift_cluster_subnet_group_name": S("ClusterSubnetGroupName"),
```
Comment (anjafr) [inline plugins/aws/resoto_plugin_aws/resource/redshift.py:322]: should this result in a `dependant_node` to the corresponding instance of `AwsEc2Subnet`?

File: plugins/aws/resoto_plugin_aws/collector.py (hunk 1 above)
Comment (meln1k) [reply plugins/aws/resoto_plugin_aws/collector.py:25]: I'm on the latest master, not sure what is newer :) 

Regarding the list, I think the order does not matter.

File: plugins/aws/resoto_plugin_aws/resource/redshift.py [hunk 6]
```diff
@@ -411,6 +413,21 @@ class AwsRedshiftCluster(AwsResource):
...
+        if self.redshift_vpc_id:
+            builder.dependant_node(self, reverse=True, clazz=AwsEc2Vpc, id=self.redshift_vpc_id)
+
+        for vsg in self.redshift_vpc_security_groups:
+            if vsg.vpc_security_group_id:
+                builder.dependant_node(self, reverse=True, clazz=AwsEc2SecurityGroup, id=vsg.vpc_security_group_id)
+
+        for role in self.redshift_iam_roles:
+            if role.iam_role_arn:
+                builder.dependant_node(self, reverse=True, clazz=AwsEc2Vpc, arn=role.iam_role_arn)
```
Comment (anjafr) [inline plugins/aws/resoto_plugin_aws/resource/redshift.py:426]: is it really `AwsEc2Vpc` for `role in self.redshift_iam_roles`?

**Review by anjafr — APPROVED** [review]
LGTM :)

**Pull request merged**